package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +541,77 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
-        if not iterable(self.weights) and self.weights == "mass":
+
+        # check weights type
+
+        if self.weights == 'mass':
             pass
-        else:
+        elif self.weights is None:
```

Comment:
I would try to get rid of all the `pass` clauses as this will also simplify testing.